tetragonpod/api/v1alpha1/groupversion_info.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2023 Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package v1alpha1 contains API Schema definitions for the cilium.io v1alpha1 API group
+// +kubebuilder:object:generate=true
+// +groupName=cilium.io.tetragon.cilium.io
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/scheme"
+)
+
+var (
+	// GroupVersion is group version used to register these objects
+	GroupVersion = schema.GroupVersion{Group: "cilium.io.tetragon.cilium.io", Version: "v1alpha1"}
```

Comment:
group name should be https://github.com/cilium/tetragon/blob/main/pkg/k8s/apis/cilium.io/register.go#L7

Reply:
Kubebuilder is giving Group value from group and domain flags, together `group.domain`.

While I was setting up kubebuilder using `kubebuilder init` command, I provided domain to be `tetragon.cilium.io` as suggested and when created the API I using `kubebuilder create api` provided `--group cililum.io`, as a result of which it is generating the group to be `cilium.io.tetragon.cilium.io` (the issue above).

Now, when I am changing things in `api/v1alpha1/groupversion_info.go`, it is not reflecting the changes in CRD yaml when running `make manifests`, Should I change the things in [Project config](https://github.com/prateek041/tetragon/blob/f02a0ee440967b3151b675a5aceca39fb8e616e5/tetragonpod/PROJECT) of my project ? It clearly suggests me not to. [ref](https://book.kubebuilder.io/reference/project-config.html)

@michi-covalent

Reply:
I have fixed the main CRD yaml, by changing the comment at [here](https://github.com/prateek041/tetragon/blob/f02a0ee440967b3151b675a5aceca39fb8e616e5/tetragonpod/api/v1alpha1/groupversion_info.go#L19). This fixed the issue in the main CRD, I still see that [rbac yaml filees](https://github.com/prateek041/tetragon/tree/f02a0ee440967b3151b675a5aceca39fb8e616e5/tetragonpod/config/rbac) still use old group and Kind in many places. how do I fix that ?

Reply:
the worst case you can just search and replace.

Reply:
fixed